app/helpers/hyku_addons/api_works_helper.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Work related API helper methods
+module HykuAddons
+  module ApiWorksHelper
+    def creator_from_json(creator_json)
+      creator_hash = JSON.parse(creator_json || "{}")
+
+      return [] if creator_hash.blank?
+
+      key = "creator_institutional_email"
+
+      creator_hash.map do |creator|
+        if creator[key].blank?
+          creator.delete(key)
+        else
+          user = find_by_user_email(creator[key])
+          creator.delete(key) unless user.present? && user.display_profile
```

Comment:
This looks much clearer now, but there i a small improvement we can make. Instead of adding in another method to find the user, then check for the object and a value on it, we can move all of that logic to the method. 

```ruby
    # ...
    creator.delete(key) unless user_display_profile?(creator[key])
    # ....

    def user_display_profile?(email)
        user = User.find_by(email: email)

        user.present? && user.display_profile
    end
````
But now that we have it condensed to a single line, we can improve it once more... 
```ruby
    creator_hash.map do |creator|
        if creator[key].blank? 
            creator.delete(key)
        elsif !user_display_profile?(creator[key])
            creator.delete(key)
        end
    end
```
We can now see that we have the same statment being run from two conditions, so there is another improvement we can make: 
```ruby
    creator_hash.map do |creator|
        creator.delete(key) if creator[key].blank? || !user_display_profile?(creator[key])
    end
```
But, its not great using an inline `if` like this for a single line, when we can invert the condition and use a guard statement instead: 
```ruby
    creator_hash.map do |creator|
         next if creator[key].present? && user_display_profile?(creator[key])
 
         creator.delete(key)
    end
```
There is one downside to this and that is because the guard clause uses an `&&`, both sides of the condition will need to be evaluated, which means we are going to do a db query for each iteration. This could potentially cause issues with larger sets of user emails. 

By using an `||` if the left side of the condition is true, then we do not need to move onto the left side: 
```ruby
    creator_hash.map do |creator|
         # Use unless to avoid querying the database for each user
         next unless creator[key].present? || !user_display_profile?(creator[key])
 
         creator.delete(key)
    end
```

Reply:
Should the last one be 
`next unless creator[key].blank? || !user_display_profile?(creator[key])`

Reply:
Quite possibly. Why do you think that? You know the end goal of this code block better than I do. 

So the original was:
```ruby
        if creator[key].blank?
          creator.delete(key)
        else
          user = find_by_user_email(creator[key])
          creator.delete(key) unless user.present? && user.display_profile
        end
```
and we ended up with: 
```ruby
    creator_hash.map do |creator|
         next unless creator[key].present? || !user_display_profile?(creator[key])
 
         creator.delete(key)
    end
```
So if we want to ignore the creator if their `creator_institutional_email` is available, or it doesn't relate to a user in the system, whilst also avoiding querying the db every iteration. 

```ruby
    creator_hash.map do |creator|
         next if creator[key].present? || !user_display_profile?(creator[key])
 
         creator.delete(key)
    end
```